src/SFML/Graphics/Font.cpp

```diff
@@ -510,17 +510,20 @@ bool Font::isSmooth() const
 ////////////////////////////////////////////////////////////
 Font& Font::operator=(const Font& right)
 {
-    Font temp(right);
+    if (this != &right)
+    {
+        Font temp(right);
 
-    std::swap(m_fontHandles, temp.m_fontHandles);
-    std::swap(m_isSmooth, temp.m_isSmooth);
-    std::swap(m_info, temp.m_info);
-    std::swap(m_pages, temp.m_pages);
-    std::swap(m_pixelBuffer, temp.m_pixelBuffer);
+        std::swap(m_fontHandles, temp.m_fontHandles);
+        std::swap(m_isSmooth, temp.m_isSmooth);
+        std::swap(m_info, temp.m_info);
+        std::swap(m_pages, temp.m_pages);
+        std::swap(m_pixelBuffer, temp.m_pixelBuffer);
 
 #ifdef SFML_SYSTEM_ANDROID
-    std::swap(m_stream, temp.m_stream);
+        std::swap(m_stream, temp.m_stream);
 #endif
+    }
```

Comment:
What's the motivation for this? I ran clang-tidy locally with these two new checks and it didn't recommend I change anything about `Font::operator=`.

Reply:
Checking for self-assignment is considered a [good thing](https://isocpp.org/wiki/faq/assignment-operators).

Reply:
That seems to be outside of the scope of this PR

Reply:
> That seems to be outside of the scope of this PR

OK, np, will revert